packages/qwik/src/core/render/dom/visitor.ts

```diff
@@ -441,6 +441,8 @@ export const diffVnode = (
         }
         if (prop === 'class') {
           newValue = serializeClassWithHost(newValue, currentComponent);
+        } else if (prop === 'className') {
```

Comment:
Don't we already auto convert className to class and print a warning? Why is that not enough?

You also need to merge both when both are present if you do this.